### PR TITLE
fix: dont use relative paths, use webpack alias instead

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

* Changes how `getEnabledPlugins` works in `@strapi/strapi` where if you pass `{ client: true }` you get the module style import instead of a direct path, again, because we're using a compiler this is typically what we'd expect.
* removes duplicate ways of generating the `plugins.js` file in the `@strapi/admin` package so it's easier to understand and the cli and package behave the same – the package also behaves the same between prod / dev (previously it was _slightly_ different)
* Loosens the `include` array because it's not actually necessary since our `entry` is very specific
* Adds aliases for dev mode for the library so our packages look at the src and everything is _hot reload_ 🐎 

### Why is it needed?

* Both locally and production we were aliasing the plugins that are installed e.g. `@strapi/plugin-upload` relatively, locally this wasn't the worst however we could/should be using the `alias` property of our development webpack configuration, however we really shouldn't be aliasing to the `node_modules` folder and around in production because we're bypassing package-managers and the compiler which can be a source of bugs.
* Using such a strict `include` array was most likely unnecessary because our entry point was _very_ accurate, this actually lead to us having other settings required to be changed e.g. `symlink` which is _typically_ fine to be `true` leading to a more complex webpack configuration.

### How to test it?

* Build the admin panel either in a strapi-app or locally
* Develop the admin panel either in a strapi-app or locally

### Related issue(s)/PR(s)

* resolves CONTENT-1806
* unblocks CONTENT-1748

### CODE REVIEW

File: packages/core/admin/scripts/build.js [hunk 1]
```diff
@@ -24,18 +18,20 @@ const buildAdmin = async () => {
...
-  const plugins = getPluginToInstallPath(PLUGINS_TO_INSTALL);
-  const allPlugins = { ...corePlugins, ...plugins };
-  const pluginsPath = getPluginsPath();
+  const plugins = getPlugins([
+    '@strapi/plugin-content-type-builder',
+    '@strapi/plugin-email',
+    '@strapi/plugin-upload',
+    '@strapi/plugin-i18n',
+    '@strapi/plugin-users-permissions',
+  ]);
```
Comment (gu-stav) [inline packages/core/admin/scripts/build.js:27]: I suppose these are the core-plugins that are installed always? Could you add a comment on that?

File: packages/core/admin/utils/create-cache-dir.js [hunk 2]
```diff
@@ -76,12 +85,35 @@ async function createCacheDir({ appDir, plugins }) {
...
+    .filter(([, plugin]) => {
+      try {
+        const isLocalPluginWithLegacyAdminFile = fs.existsSync(
+          path.resolve(`${plugin.pathToPlugin}/strapi-admin.js`)
+        );
+
+        if (!isLocalPluginWithLegacyAdminFile) {
+          const isModulewithLegacyAdminFile = require.resolve(
+            `${plugin.pathToPlugin}/strapi-admin.js`
+          );
```
Comment (gu-stav) [inline packages/core/admin/utils/create-cache-dir.js:98]: Do you think it would make sense to add a comment about what the difference about these two ways is? To me it isn't really obvious immediately.

File: packages/core/admin/utils/get-plugins.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+      const { name, strapi } = require(join(directory, 'package.json'));
+
+      /**
+       * this will remove any of our packages that are
+       * not actually plugins for the application
+       */
+      if (!strapi) {
+        return null;
+      }
```
Comment (gu-stav) [inline packages/core/admin/utils/get-plugins.js:45]: Should this throw a meaningful error, if strapi is not part of package.json e.g. for plugin developers instead?

File: packages/core/admin/webpack.config.dev.js [hunk 4]
```diff
@@ -47,17 +47,32 @@ module.exports = () => {
...
+          return acc;
+        }, {}),
+      },
+    },
+
+    plugins: [
+      ...config.plugins,
+      analyzeBundle && new BundleAnalyzerPlugin(),
+      analyzeDuplicateDependencies === 'true' && new DuplicateReporterPlugin(),
+      process.env.REACT_REFRESH !== 'false' && new ReactRefreshWebpackPlugin(),
```
Comment (gu-stav) [inline packages/core/admin/webpack.config.dev.js:73]: Why did you move that to the `dev` config? Are plugin developers able to still use it?

**Review by gu-stav — COMMENTED** [review]
Code looks good to me. Will do some functional testing later!

File: packages/core/admin/utils/get-plugins.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+const { join, resolve, sep, posix } = require('path');
+const fs = require('fs');
+// eslint-disable-next-line import/no-extraneous-dependencies
+const glob = require('glob');
+
+const getPlugins = (pluginsWhitelist) => {
```
[the remark this answers is not in the document]
Comment (joshuaellis) [reply packages/core/admin/utils/get-plugins.js]: Definitely, kind of new it was an outdated term when i wrote it but couldn't think of the better word at the time, thanks!

File: packages/core/admin/webpack.config.dev.js (hunk 4 above)
Comment (joshuaellis) [reply packages/core/admin/webpack.config.dev.js:73]: ahhh is that why it's there for the `watchAdmin` stuff, I can put this back then, thanks 🧠

File: packages/core/admin/utils/get-plugins.js (hunk 3 above)
Comment (joshuaellis) [reply packages/core/admin/utils/get-plugins.js:45]: No i don't, because this is only used in our internal development, not when it's published :)

File: packages/core/admin/utils/create-cache-dir.js (hunk 2 above)
Comment (joshuaellis) [reply packages/core/admin/utils/create-cache-dir.js:98]: 👍🏼

**Review by Marc-Roig — APPROVED** [review]

**Review by gu-stav — APPROVED** [review]
Works for me. Well done!
